tests/manage/monitoring/conftest.py

```diff
@@ -130,30 +130,45 @@ def prometheus_log(info, alert_list):
         )
         logging_thread.start()
 
-        result = operation()
-        if measure_after:
-            start_time = time.time()
-        passed_time = time.time() - start_time
-        if minimal_time:
-            additional_time = minimal_time - passed_time
-            if additional_time > 0:
-                logger.info(f"Starting {additional_time}s sleep for the purposes of measurement.")
-                time.sleep(additional_time)
-        stop_time = time.time()
-        info['run'] = False
-        logging_thread.join()
-        results = {
-            'start': start_time,
-            'stop': stop_time,
-            'result': result,
-            'metadata': metadata,
-            'prometheus_alerts': alert_list,
-            'first_run': True,
-        }
-        logger.info(f"Results of measurement: {results}")
-        with open(result_file, 'w') as outfile:
-            logger.info(f"Dumping results of measurement into {result_file}")
-            json.dump(results, outfile)
+        try:
+            result = operation()
+        except Exception as ex:
```

Comment:
this is too wide. Is there any specific exception to catch instead?

Reply:
This is too wide on purpose, because `operation` is mandatory argument of this ` measure_operation`, and so I can't assume anything about the possible exception here. On the other hand I need to make sure that in case of any problem, the alert harvesting thread is shutdown properly.